Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -127,11 +127,16 @@ public AsyncFunctionAssertions(Func<TTask> subject, IExtractExceptions extractor
     {
         Type expectedType = typeof(TException);
 
-        Execute.Assertion
+        bool success = Execute.Assertion
             .ForCondition(Subject is not null)
             .BecauseOf(because, becauseArgs)
             .FailWith("Expected {context} to throw exactly {0}{reason}, but found <null>.", expectedType);
 
+        if (!success)
+        {
+            return new ExceptionAssertions<TException>(Array.Empty<TException>());
```

Comment:
Wondering why this "early return" is accepted here.
In the past I was requested to NOT early return.

Reply:
I agree with you. It makes the code more difficult to read.

Reply:
No, I (myself!!!) don't think so, but it is style you (!!) prefer.
So, I was wondering why do you approve such changes, @dennisdoomen ;)

Reply:
Because we're just humans and sometimes miss things...